Update `CONTRIBUTING.md` to include `Simplecov` (#2035)
rubocop/rubocop-rspec: .github/CONTRIBUTING.md

$ mdformat . --number
```

### Test Coverage - Line and Branch

We are using [Simplecov](https://github.com/colszowka/simplecov) to track test coverage.

> It is included and reported when you run `bundle exec rake`.

Reviewer: also when you run `bundle exec rspec`.

To view the coverage report, open the `coverage/index.html` file in your browser. Here is a quick generic command to find and open the file on macOS:

```console

> $ open $(find . -type f -path "*/coverage/index.html" -print -quit)

Reviewer: This simpler version has never failed me:
```suggestion
$ open coverage/index.html
```
Plus, if the file doesn’t exist (before first spec run) you get a usable error message.

Author: Fair enough, I aliased this to work across nested repos at some point and never looked back.

`alias cov=...`

```
If you have unreachable lines, you can add `# :nocov` around those lines. The code itself or a comment should explain why the line is unreachable.

Example:
```ruby
# :nocov:
raise ArgumentError("Unsupported style :#{style}")
# :nocov:
```

This can happen for a few reasons:
1. When you handle config with a case statement and there is no else block
2. When matching with a node pattern even when you handle all cases: all other node types will be excluded before reaching your handler, because the node pattern will not match them.

> You will need full line and branch coverage to merge. This has helped prevent and detect errors and edge cases.

Reviewer: ```suggestion
You will need full line and branch coverage to merge. This helps detect edge cases and prevent errors.
```

$ mdformat . --number
```

### Test Coverage - Line and Branch

We are using [Simplecov](https://github.com/colszowka/simplecov) to track test coverage.

It is included and reported when you run `bundle exec rake` or `bundle exec rspec`.

To view the coverage report, open the `coverage/index.html` file in your browser. 

E.g. on macOS:
```console
$ open coverage/index.html
```
If you have unreachable lines, you can add `# :nocov` around those lines. The code itself or a comment should explain why the line is unreachable.

Example:
```ruby
# :nocov:
raise ArgumentError("Unsupported style :#{style}")
# :nocov:
```

This can happen for a few reasons, including:

> 1. When you handle config with a case statement and there is no else block

Reviewer: ```suggestion
1. When you handle config with a case statement and there is no else block.
```
